AI/MMAI/CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+cmake_minimum_required(VERSION 3.24)
+
+set(MMAI_FILES
+  BAI/base.cpp
+  BAI/base.h
+  BAI/router.cpp
+  BAI/router.h
+  BAI/model/ScriptedModel.h
+  BAI/model/ScriptedModel.cpp
+  BAI/model/NNModel.h
+  BAI/model/NNModel.cpp
+
+  BAI/v13/BAI.cpp
+  BAI/v13/BAI.h
+  BAI/v13/action.cpp
+  BAI/v13/action.h
+  BAI/v13/attack_log.h
+  BAI/v13/battlefield.cpp
+  BAI/v13/battlefield.h
+  BAI/v13/encoder.cpp
+  BAI/v13/encoder.h
+  BAI/v13/global_stats.cpp
+  BAI/v13/global_stats.h
+  BAI/v13/hex.cpp
+  BAI/v13/hex.h
+  BAI/v13/hexaction.h
+  BAI/v13/hexactmask.h
+  BAI/v13/links.h
+  BAI/v13/player_stats.cpp
+  BAI/v13/player_stats.h
+  BAI/v13/render.cpp
+  BAI/v13/render.h
+  BAI/v13/stack.cpp
+  BAI/v13/stack.h
+  BAI/v13/state.cpp
+  BAI/v13/state.h
+  BAI/v13/supplementary_data.cpp
+  BAI/v13/supplementary_data.h
+  BAI/v13/util.cpp
+  BAI/v13/util.h
+
+  schema/schema.h
+  schema/v13/constants.h
+  schema/v13/schema.h
+  schema/v13/types.h
+  schema/v13/util.h
+
+  MMAI.h
+  StdInc.h
+  common.h
+)
+
+option(ENABLE_MMAI_TEST "Compile tests" OFF)
+option(ENABLE_MMAI_STRICT_LOAD "Disable MMAI fallback during model load and throw an error instead" OFF)
+
+set(MMAI_INCLUDES ${CMAKE_CURRENT_SOURCE_DIR})
+
+add_definitions(-DUSING_ONNX=1)
+assign_source_group(${MMAI_FILES})
+
+if(ENABLE_STATIC_LIBS)
+  add_library(MMAI STATIC ${MMAI_FILES})
+else()
+  add_library(MMAI SHARED ${MMAI_FILES} main.cpp StdInc.cpp)
+  install(TARGETS MMAI RUNTIME DESTINATION ${AI_LIB_DIR} LIBRARY DESTINATION ${AI_LIB_DIR})
+
+  # This is needed to allow MMAI to link against BattleAI
+  # For windows, this is handled at runtime instead via LoadLibraryExW(..., LOAD_WITH_ALTERED_SEARCH_PATH)
+  if(APPLE)
+    set_target_properties(MMAI PROPERTIES INSTALL_RPATH "@loader_path")
+  elseif(UNIX)
+    set_target_properties(MMAI PROPERTIES INSTALL_RPATH "$ORIGIN")
+  endif()
+endif()
+
+target_link_libraries(MMAI PRIVATE vcmi)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  set(ONNXRUNTIME_ROOT "/opt/onnxruntime" CACHE PATH "Path to onnxruntime containing lib/onnxruntime.so and include/onnxruntime_cxx_api.h")
```

Comment:
Maybe add check if onnxruntime is available in system and only if not use it from downloaded file.

There are already some distributions with packages, e.g.:
- [Debian 13](https://packages.debian.org/trixie/libonnxruntime-dev)
- [Ubuntu 24.10](https://launchpad.net/ubuntu/oracular/+package/libonnxruntime-dev) (should be in next LTS as well)
- [Arch](https://archlinux.org/packages/extra/x86_64/onnxruntime/)
- [Fedora](https://packages.fedoraproject.org/pkgs/onnxruntime/onnxruntime-devel/)